### ISSUE TITLE
Bare imports should always be executed first

### ISSUE THREAD

**Issue by arcanis** [evanw/esbuild#2598]
Consider the [following code](https://hyrious.me/esbuild-repl/?version=0.15.10&mode=build&modules=%5B%5B%22main.js%22%2C%22import+%27.%2Fpolyfill%27%3B%5Cnimport+%7Bfoo%7D+from+%27.%2Fuser-lib%27%3B%5Cn%5Cnfoo%28%29%3B%5Cn%22%2C1%5D%2C%5B%22polyfill.js%22%2C%22Object.somePolyfilledFunction+%3D+%28%29+%3D%3E+%7B%7D%3B%22%2C0%5D%2C%5B%22user-lib.js%22%2C%22Object.somePolyfilledFunction%28%29%3B%5Cn%5Cnexport+async+function+foo%28%29+%7B%5Cn%5Ctreturn+import%28%27.%2Flazy-chunk%27%29%3B%5Cn%7D%22%2C0%5D%2C%5B%22lazy-chunk.js%22%2C%22import+%7Bfoo%7D+from+%27.%2Fuser-lib%27%3B%5Cn%5Cnfoo%28%29%3B%22%2C0%5D%5D&buildOptions=%7B%22bundle%22%3Atrue%2C%22format%22%3A%22esm%22%2C%22splitting%22%3Atrue%7D):

```ts
import './polyfill';
import {foo} from './user-lib';

foo();
```

Esbuild inlines `./polyfill`, causing the file to turn into:

```
import {
  foo
} from "./chunk-EN3LV7DE.js";

// polyfill.js
Object.somePolyfilledFunction = () => {
};

// main.js
foo();
```

This is invalid, as it means that the polyfill won't be available inside `chunk-EN3LV7DE`.

Esbuild should at least assume that the bare import, having no imported symbols, must have a significance in terms of side-effects and ensure that it gets evaluated at the right time.

**Comment by hyrious** [comment]
#399

Note that this problem also exists in Rollup.

**Comment by arcanis** [comment]
I checked #399 but since it mentioned multiple entry points I wasn't sure if this was a different issue or not 🤔

**Comment by hanayashiki** [comment]
seems you can't specify the order of execution of ecma modules per spec

**Comment by hanayashiki** [comment]
you can keep your polyfill while delay the run ofpiece of code that depends on it (e.g. lazy evaluation)

**Comment by arcanis** [comment]
Yes, that's what I currently do, but it's not ideal: the browser has to finish executing polyfills before even starting to fetch the actual application. It adds a small delay and indirection; I'm a little worried a browser will eventually show a white flash on reload if the delay is long enough.
